### ISSUE TITLE
Error in Train Step

### ISSUE THREAD

**Issue by prasad579** [dotnet/machinelearning-modelbuilder#423]
Problem encountered on https://dotnet.microsoft.com/learn/ml-dotnet/get-started-tutorial/train
Operating System: windows

Hello,
I am new to machine learning and followed your tutorial steps to learn.But i was stuck with a error in train module.
![image](https://user-images.githubusercontent.com/31720292/71902480-2ab46480-3188-11ea-911a-4697ca1a4cb5.png)
Here I am n ot seeing any errors in output window too.Please help me with the solution.
Operating System - Windows 10

**Comment by Chaopeng-Wu** [comment]
can i see you step.2 data panel?

**Comment by prasad579** [comment]
sure.Here it  is

![image](https://user-images.githubusercontent.com/31720292/71956239-a0fca980-3210-11ea-99da-db166abbb507.png)

**Comment by LittleLittleCloud** [comment]
Does the training fail at once, or it fails after 10s?

**Comment by beccamc** [comment]
@prasad579 Are you still having problems? Did this fail right away? Or after 10 seconds?

**Comment by mkArtak** [comment]
Hi.
I'm experiencing the same issue.
@beccamc, to answer your question, the training fails right away - no delays.
What information is needed to investigate this, I'm ready to provide it.

**Comment by mkArtak** [comment]
Here is the VS information:
![image](https://user-images.githubusercontent.com/7347708/74079346-f7fdc600-49ea-11ea-88b0-7deee148f318.png)

And here is the ML.Net Model Builder extension info:
![image](https://user-images.githubusercontent.com/7347708/74079356-119f0d80-49eb-11ea-8246-8c0db88d5ba5.png)

I'm running on a Win 10 x64 machine.

**Comment by mkArtak** [comment]
Hi @beccamc. Is this something being worked on? Or is this scenario not a priority any more?

**Comment by JakeRadMSFT** [comment]
@mkArtak can you try with latest verion? We're not able to reproduce the issue.

**Comment by mkArtak** [comment]
I have the latest available Community edition installed:
![image](https://user-images.githubusercontent.com/7347708/75744401-0036e180-5cc9-11ea-9101-6f2f644270bc.png)

And I still observe the same issue.

**Comment by beccamc** [comment]
@mkArtak Sorry for the long delay in response on this issue. Are you trying with the sample dataset [Wikipedia detox dataset](https://raw.githubusercontent.com/dotnet/machinelearning/master/test/data/wikipedia-detox-250-line-data.tsv), or your own dataset?

**Comment by mkArtak** [comment]
It was my own data set
[data.txt](https://github.com/dotnet/machinelearning-modelbuilder/files/4479428/data.txt)

**Comment by beccamc** [comment]
@mkArtak Thanks for sharing the data! The dataset has a small error that blocks training. On line 17 there is a space instead of a tab between `830 `and `25`. Changing to a tab fixes the parsing error shown during training. 

Even with that fix the dataset will still run out of time training. There are two reasons for this. 30 rows of data is a fairly small sample and difficult to train against (ML.Net will split the set into 80% training and 20% testing). Additionally, it looks like the scenario you're trying to accomplish is more closely aligned to forecasting - predict future cases based on historical data.

Unfortunately we don't support forecasting in Model Builder yet. You can check out this [tutorial](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/time-series-demand-forecasting) in the ML.NET docs.  If you want to stay in Model Builder you could try adding the additional data from March and April to your dataset and see if that is enough data to train on. I'm guessing you won't get a very accurate model since you're really looking for a forecasting model. 

Let me know if I've misinterpreted your dataset and scenario. I'll pass along to our PM a request for forecasting in Model Builder.

**Comment by mkArtak** [comment]
Thanks @beccamc.
I understand that the data set was super small, it was not intended for a production use in any way, rather than a simple experimentation.

The only actionable feedback for you from this thread is to potentially expose a more relevant error message to the users, when the training fails due to data error.

**Comment by beccamc** [comment]
@mkArtak Make sense! I've updated the error shown from timeout to provide more guidance. I recently improved our text loader and it does a way better job with your data file (available in the next release)

![image](https://user-images.githubusercontent.com/9122518/79615650-24535400-80b8-11ea-8942-d012fa055fcc.png)



We have on our backlog to show better messages to point users the exact problem row - #416. 

Thanks for providing additional info to help!
